### PR TITLE
combine some SQL update statements into fewer

### PR DESCRIPTION
**Pull request by mtbc**
Attempts to reduce the number of times the same tables are updated, to decrease upgrade time. Read the new SQL carefully in case of mistakes. Try the upgrade with a database with data in it, both from 5.0 and from early 5.1.

### CODE REVIEW

**Comment by joshmoore** [comment]
@sbesson : so now the question -- a `release-upgrade` like build for `merge-*`?

**Comment by joshmoore** [comment]
Oops, this isn't mergeable.

**Comment by joshmoore** [comment]
Waiting on the performance of https://ci.openmicroscopy.org/job/OMERO-5.1-merge-upgrade/VERSION=5.0.0,label=dogfish/198/console

**Comment by joshmoore** [comment]
> new SQL carefully in case of mistakes.

Would code-generation of the upgrades have been useful? Would it still make sense to compare?

File: sql/psql/OMERO5.1DEV__17/OMERO5.0__0.sql [hunk 1]
```diff
@@ -1649,10 +1668,6 @@ ALTER TABLE transmittancerange
 
 ALTER TABLE experimentergroup ADD COLUMN ldap BOOL NOT NULL DEFAULT false;
 
--- Set "ldap" value for each group to be false
-
-UPDATE experimentergroup SET ldap = false;
```
Comment (joshmoore) [inline sql/psql/OMERO5.1DEV__17/OMERO5.0__0.sql]: Where does this removal come from?

Comment (joshmoore) [inline sql/psql/OMERO5.1DEV__17/OMERO5.0__0.sql]: Ah, the `NOT NULL DEFAULT`.

**Comment by mtbc** [comment]
I think the changes are small enough that by-hand-and-eye suffices for this PR but, in general, certainly for anything bigger, a script for generating the updates sounds like a fine idea.

**Comment by mtbc** [comment]
(In my previous job where we'd usually drop the indices for the upgrade, we weren't doing the UPDATE-conflation that this PR does; it's plausible that either approach suffices.)

**Comment by sbesson** [comment]
As per https://ci.openmicroscopy.org/job/OMERO-5.1-merge-upgrade/VERSION=5.0.0,label=dogfish/198/, the DB ugprade took 2h30 as opposed to 23hr from the latest OMERO on the same node (see https://ci.openmicroscopy.org/job/OMERO-5.1-release-upgrade/VERSION=5.0.0,label=dogfish/17/). Assuming the code changes make sense, the performance improvement is certainly significant enough to merge this PR.

**Comment by joshmoore** [comment]
I've read over this multiple times and have found no issues. (Considering the fiddliness of the code, I may have overlooked something). I'm merging with the assumption that if there are any wrong defaults here, they will be found out during testing. Thanks, @mtbc.

**Pull request merged**